pkg/scheduler/eventhandlers.go

```diff
@@ -522,6 +528,26 @@ func addAllEventHandlers(
 	return nil
 }
 
+// podSchedulingPropertiesChange interprets the update of a pod and returns corresponding UpdatePodXYZ event(s).
+func podSchedulingPropertiesChange(newPod *v1.Pod, oldPod *v1.Pod) []framework.ClusterEvent {
+	var events []framework.ClusterEvent
+
+	if labelsChanged(newPod.ObjectMeta, oldPod.ObjectMeta) {
+		events = append(events, queue.PodLabelChange)
+	}
+
+	if podResourceRequestChanged(newPod, oldPod) {
+		events = append(events, queue.PodRequestChange)
+	}
+
+	if len(events) == 0 {
+		events = append(events, queue.AssignedPodUpdate)
+	}
```

Comment:
We have to tell other updates as `AssignedPodUpdate` because custom plugins may want to subscribe to other kinds of updates in Pods. 
So, this PR itself doesn't contribute to a reduction of the number of events registered in inFlightPods. But, after https://github.com/kubernetes/kubernetes/pull/122629, this PR does.

Reply:
Will we miss some combined pod events here, like pod annotations & pod label updates, as well as pod status.
We should make sure it's compatible.

Reply:
I don't think a combined event would be a problem here.
Let's say one Pod update contains the update in annotations & label - the update will be interpreted as `PodLabelChange`, and this `PodLabelChange` will be matched with Pod/Update as well = QHint registered for Pod/Update will be called with PodLabelChange event.